tests/test_models.py

```diff
@@ -1,37 +1,329 @@
-        """ It should always be true """
-        self.assertTrue(True)
+    def test_create_a_promotion(self):
+        """It should Create a promotion and assert that it exists"""
+        today = date.today()
+        tomorrow = today + timedelta(1)
+        promo = Promotion(name="20% Off", start_date=today, end_date=tomorrow, whole_store=True, message="This is a test!", promotion_changes_price=True)
+        self.assertEqual(str(promo), "<Promotion 20% Off id=[None]>")
+        self.assertTrue(promo is not None)
+        self.assertEqual(promo.id, None)
+        self.assertEqual(promo.name, "20% Off")
+        self.assertEqual(promo.start_date, today)
+        self.assertEqual(promo.end_date, tomorrow)
+        self.assertEqual(promo.is_active(), True)
+        self.assertEqual(promo.whole_store, True)
+        self.assertEqual(promo.message, "This is a test!")
+        self.assertEqual(promo.promotion_changes_price, True)
+        promo = Promotion(name="20% Off", start_date=today, end_date=tomorrow, whole_store=False, message="This is a test!", promotion_changes_price=False)
+        self.assertEqual(promo.whole_store, False)
+        self.assertEqual(promo.promotion_changes_price, False)
+
+    def test_add_a_promotion(self):
+        """It should Create a promotion and add it to the database"""
+        today = date.today()
+        tomorrow = today + timedelta(1)
+        promos = Promotion.all()
+        self.assertEqual(promos, [])
+        promo = Promotion(name="20% Off", start_date=today, end_date=tomorrow, whole_store=True, message="This is a test!", promotion_changes_price=True)
+        self.assertTrue(promo is not None)
+        self.assertEqual(promo.id, None)
+        promo.create()
+        # Assert that it was assigned an id and shows up in the database
+        self.assertIsNotNone(promo.id)
+        promos = Promotion.all()
+        self.assertEqual(len(promos), 1)
+
+    def test_read_a_promotion(self):
+        """It should Read a Promotion"""
+        promo = PromoFactory()
+        logging.debug(promo)
+        promo.id = None
+        promo.create()
+        self.assertIsNotNone(promo.id)
+        # Fetch it back
+        found_promo = Promotion.find(promo.id)
+        self.assertEqual(found_promo.id, promo.id)
+        self.assertEqual(found_promo.name, promo.name)
+        self.assertEqual(found_promo.start_date, promo.start_date)
+
+    def test_update_a_promotion(self):
+        """It should Update a Promotion"""
+        promo = PromoFactory()
+        logging.debug(promo)
+        promo.id = None
+        promo.create()
+        logging.debug(promo)
+        self.assertIsNotNone(promo.id)
+        # Change it an save it
+        promo.end_date = date.today() + timedelta(2)
+        original_id = promo.id
+        promo.update()
+        self.assertEqual(promo.id, original_id)
+        self.assertEqual(promo.end_date, date.today() + timedelta(2))
+        # Fetch it back and make sure the id hasn't changed
```

Comment:
If we use "has_been_extended", we need update "has_been_extended" here as well?

Reply:
good catch!